core/runtime/src/main/java/io/quarkus/runtime/Quarkus.java

```diff
@@ -67,18 +72,17 @@ public static void run(Class<? extends QuarkusApplication> quarkusApplication, B
             //we already have an application, run it directly
             Class<? extends Application> appClass = (Class<? extends Application>) Class.forName(Application.APP_CLASS_NAME,
                     false, Thread.currentThread().getContextClassLoader());
-            Application application = appClass.getDeclaredConstructor().newInstance();
+            MethodHandle constructor = MethodHandles.lookup().findConstructor(appClass, MethodType.methodType(void.class));
+            Application application = (Application) constructor.invoke();
```

Comment:
A minor nit here, you should be able to use `invokeExact`, but if not, it's not a big deal. And the `unchecked` warnings suppression shouldn't be needed since this is a method handle call (if it is, your IDE might be misconfigured).

Reply:
I actually tried it, but the JVM didn't like because `io.quarkus.runner.ApplicationImpl` does not exactly match `io.quarkus.runtime.Application`

Reply:
Makes sense!